build.gradle

```diff
@@ -93,13 +96,73 @@ tasks.named('build') {
 
 test {
     useJUnitPlatform()
+
+    testLogging {
```

Comment:
These configs add logging to our test runs, which has been helpful in trying to debug. We can leave them in or modify them or rip them out, but I think we should look at improving what we output when tests are running.

Reply:
Nice, we can leave it for now and if it gets noisy or creates any issue, we can reconsider it later

Reply:
Actually we can debug the tests by adding flags --debug and/or --stacktrace on local or in CI, so is it okay if we remove this for now and keep the focus of this PR on adding tests for collections only?

Reply:
We can't add --debug to the CI though because it can leak information. So if the logging is required, we can add this block but for now, the CI will tell us which test failed with what condition.